### DISCUSSION TITLE
Is there a way to resize animated images?

### DISCUSSION THREAD

**Discussion by Nekidev** [codingjoe/django-pictures#111] [Q&A] [answered]
Basically the title. If I resize an animated GIF the results are all static. Is there a way to keep them animated and resized?

**Comment by codingjoe** [comment]
Hi @Nekidev,

Thanks, an excellent question. Honestly, I don't know. I did just browse through the Pillow documentation, but didn't find any remakes on it NOT working. Did you give it a try?

If Pillow supports it, it might already work for this package. If not, I'd be eager to add support for it.

Cheers!
Joe

**Reply by Nekidev** [reply]
Hi. I ended up using django-thumbnails which allows making custom image processors.

Pillow does not modify all frames of an animated image by default, so I needed to loop through the frames cropping each of them and then rejoined them to generate the new cropped GIF.

This is a tiny example of what I used. My version is modified to resize the gif to fit a certain aspect ratio.

```py
from PIL import Image, ImageSequence


size = 160, 90

img = Image.open("in.gif")

frames = ImageSequence.Iterator(im)


# Wrap on-the-fly thumbnail generator
def thumbnails(frames):
    for frame in frames:
        thumbnail = frame.copy()
        thumbnail = thumbnail.crop(
            (
                (img.width - size[0]) / 2,
                (img.height - size[1]) / 2,
                (img.width + size[0]) / 2,
                (img.height + size[1]) / 2
            ),
            Image.ANTIALIAS
        )
        yield thumbnail


frames = thumbnails(frames)


# Save output
om = next(frames) # Handle first frame separately
om.info = im.info # Copy sequence info
om.save(
    "out.gif",
    save_all=True,
    append_images=list(frames),  # Append all other frames to the first frame. If this is not added only the first frame is exported, which removes the GIF animation.
    duration=50,  # Duration in ms of each frame. It is ignored if it is too small
    loop=0,  # Amount of times to loop. 0 is infinite.
    optimize=False  # I haven't investigated much about this, but I believe that it merges duplicate frames into a single frame with + duration to reduce file size.
)
```

It is a modified version of [this gist](https://gist.github.com/skywodd/8b68bd9c7af048afcedcea3fb1807966)

**Reply by codingjoe** [reply, marked as answer]
Interesting, you can use this library with a custom image processor. We could even simplify the use, with a tiny code change, if that works better. In fact, while writing this answer, I did write a patch making this process super simple. For your convenience, please see #114 

I hope this helps you! Joe
